### PR TITLE
[O2B-1365] Disable GAQ for not fully covered runs

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- GAQ summary is not displayed when not all runs are fully covered by QC flags
- GAQ overview shows not covered periods

Notable changes for developers:
- Added QC flags seeders to cover corner cases of QC/GAQ summaries displays
- Reworked QC flag Types seeders to make them more meaningful

Changes made to the database:
- NA

### CODE REVIEW

File: test/public/runs/runsPerDataPass.overview.test.js [hunk 1]
```diff
@@ -165,12 +165,16 @@ module.exports = () => {
...
+        await expectInnerText(page, '#row106-globalAggregatedQuality', 'GAQ');
+
+        await goToPage(page, 'runs-per-data-pass', { queryParameters: { dataPassId: 3 } });
+        await expectInnerText(page, '#row56-globalAggregatedQuality', '0MC.R');
+        expect(await getPopoverInnerText(await page.waitForSelector('#row56-globalAggregatedQuality .popover-trigger')))
+            .to.be.equal('Missing 4 verifications');
     });
 
     it('should switch mcReproducibleAsNotBad', async () => {
+        await goToPage(page, 'runs-per-data-pass', { queryParameters: { dataPassId: 1 } });
```
Comment (martinboulais) [inline test/public/runs/runsPerDataPass.overview.test.js:177]: Same, we should avoid goToPage

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/services/qualityControlFlag/QcFlagService.js [hunk 2]
```diff
@@ -357,6 +359,12 @@ class QcFlagService {
             this._mergeIntoSummaryUnit(runDetectorSummary, runDetectorSummaryForFlagTypesClass);
         }
 
+        for (const runSummaries of Object.values(summary)) {
+            for (const runDetectorSummary of Object.values(runSummaries)) {
+                this._fillQcSummaryMissingValues(runDetectorSummary);
+            }
+        }
+
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js:367]: Why not to wrap single-for-loop with this filling method which occur in `getGaqSummary(...)` ?

File: test/public/runs/runsPerDataPass.overview.test.js [hunk 3]
```diff
@@ -165,12 +165,16 @@ module.exports = () => {
         });
         await page.waitForSelector('tr#row106 .column-CPV a .icon');
 
-        await expectInnerText(page, '#row106-globalAggregatedQuality', '67MC.R');
-        expect(await getPopoverInnerText(await page.waitForSelector('#row106-globalAggregatedQuality .popover-trigger')))
-            .to.be.equal('Missing 3 verifications');
+        await expectInnerText(page, '#row106-globalAggregatedQuality', 'GAQ');
+
+        await goToPage(page, 'runs-per-data-pass', { queryParameters: { dataPassId: 3 } });
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply test/public/runs/runsPerDataPass.overview.test.js]: fixed

File: lib/server/services/qualityControlFlag/QcFlagService.js [hunk 4]
```diff
@@ -65,14 +65,16 @@ const validateUserDetectorAccess = (userRoles, detectorName) => {
...
  * @property {number} missingVerificationsCount - number of not verified QC flags which are not discarded
  * @property {boolean} mcReproducible - states whether some Limited Acceptance MC Reproducible flag was assigned
  */
 const QC_SUMMARY_PROPERTIES = {
     badEffectiveRunCoverage: 'badEffectiveRunCoverage',
     explicitlyNotBadEffectiveRunCoverage: 'explicitlyNotBadEffectiveRunCoverage',
+    qualityNotDefinedEffectiveRunCoverage: 'qualityNotDefinedEffectiveRunCoverage',
     missingVerificationsCount: 'missingVerificationsCount',
     mcReproducible: 'mcReproducible',
 };
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js]: done

File: lib/database/repositories/QcFlagRepository.js [hunk 5]
```diff
@@ -27,6 +27,26 @@ const GAQ_PERIODS_VIEW = `
                 ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING
             ) AS \`to\`
         FROM (
+                -- Two selects for runs' timestamps (in case QC flag's eff. period doesn't start at run's start or end at run's end )
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/database/repositories/QcFlagRepository.js:30]: applied in xsalonx/gaq/O2B-1365/disable-gaq-if-detector-is-missing---with-views
